tests/rptest/services/redpanda.py

```diff
@@ -2338,6 +2346,18 @@ def broker_address(self, node):
         else:
             return None
 
+    def kerberos_broker_address(self, node):
```

Comment:
It's not obvious to me how `broker_addresses` above determines which listener it hands out. Perhaps it should incorporate this change and have a default listener? 
```python
        def broker_address(self, node, listener: str = "dnslistener"):
```
And wire that up/down to the other methods; `brokers`/`kerberos_brokers` and `brokers_list`/`kerberos_brokers_list`